src/pykiso/lib/auxiliaries/can_auxiliary.py

```diff
@@ -0,0 +1,296 @@
+        :param message_name: name of the message to receive
+        :param timeout time to wait till a message receives in seconds
+
+        :return: list of last can messages or None if no messages for this component
+        """
+        last_msg = self.can_messages.get(message_name, None)
+        time.sleep(timeout)
+        new_can_msg = self.can_messages.get(message_name, None)
+        if last_msg is None or last_msg.time_stamp != new_can_msg.time_stamp:
+            return new_can_msg
+        else:
+            return None
+
+    def wait_for_signal(
+        self,
+        message_name: str,
+        expected_signals: dict[str, any],
+        timeout: float = 0.2
+    ) -> dict[str, any]:
+        """Get the last signal of message with certain timeout in seconds.
+
+        :param message_name: name of the message to receive
+        :param signal_name: signal name in the message to receive
+        :param timeout time to wait till a message receives in seconds
+
+        :return: list of last can messages or None if no messages for this component
+        """
+        
+        t1 = time.perf_counter()
+        while time.perf_counter() - t1 < timeout:
+            expected_signals_names = expected_signals.keys()
+            last_can_msg = self.can_messages.get(message_name, None)
+            if last_can_msg is None:
+                continue
+            for signal_name in expected_signals_names:
+                if last_can_msg.signals[signal_name] != expected_signals[signal_name]:
+                    continue
+            return last_can_msg
+        
+        return None
+
+    @staticmethod
+    def _is_payload_valid(payload: Optional[bytes]) -> bool:
+        """Check that the payload is valid. The payload is valid if it is made up of
+            8 bytes and bytes 6, 7 and 8 are empty.
+
+        :param payload: payload to check validity
+
+        :return: True if the payload is valid, False if invalid.
+        """
+        return payload is not None and len(payload) == 8 and (payload[5] == payload[6] == payload[7] == 0)
+
+    def send_message(self, message: str, signals: dict[str, Any]) -> bool:
+        """Send one message, the message need to be defined in the dbc file.
+
+        :param message: name of the message to send.
+        :param signals: dict of the signals of the message and their value.
+
+        :return: True or False if the message has been successfully send or not.
+        """
+
+        for signal, value in signals.items():
+            if isinstance(value, str):
+                signals[signal] = int.from_bytes(value.encode("utf8"), byteorder="big")
+        try:
+            message = self.parser.dbc.get_message_by_name(message)
+        except KeyError:
+            raise ValueError(f"{message} is not a message defined in the DBC file.")
+        msg_to_send = self.parser.encode(message, signals)
+        self.channel.cc_send(msg_to_send[0], remote_id=msg_to_send[1])
+
+    def run_command(
```

Comment:
Why is this API reimplemented? It's already part of the DTAuxiliaryInterface

Reply:
Ok, i will remove it